ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -484,7 +493,9 @@ KEXT_STATIC int HandleVnodeOperation(
             if (isRename)
             {
                 // Prevent system services from expanding directories as part of enumeration as this tends to cause deadlocks with the kauth listeners for Antivirus software
-                if (!TryGetVirtualizationRoot(&perfTracer, context, currentVnode, pid, CallbackPolicy_UserInitiatedOnly, &root, &vnodeFsidInode, &kauthResult, kauthError))
+                if (!TryGetVirtualizationRoot(&perfTracer, context, currentVnode, pid, CallbackPolicy_UserInitiatedOnly,
+                    false, // allow deleting offline directories even if not expanded
```

Comment:
@wilbaker to double check do we want to allow deleting offline directories?

Reply:
I'm not @wilbaker, but my understanding was that we want to support the case where a root is not mountable (e.g. access to server repo is no longer available, no network connection, etc.) but the user wants to delete the whole thing simply by performing an `rm -rf` on it.

Reply:
:-) That makes sense as a use case.  @pmj was there a specific set of use cases allowed given as a guide?  I looked in the attached issue but may have missed something.

Reply:
I don't think so. I just remember that case being mentioned at some point. Possibly as far back as our original kext prototype. In fact, it might be documented in the source code there?

Reply:
@pmj I think it would be useful to document the list of offline operations we are allowing in the PR description.  
I can see why delete is allowed.  However, if someone unmounts, deletes a single file, and then remounts we'll have incorrect status.  This may be a known possibility with the Windows implemenation.

Reply:
I've commented on the original issue #182 to solicit feedback on the list, and I've added the current list to the PR's description.